INSTALL

```diff
@@ -344,19 +344,11 @@
                    work if the zlib or zlib-dynamic options are also chosen.
 
   enable-crypto-mdebug
-                   Build support for debugging memory allocated via
-                   OPENSSL_malloc() or OPENSSL_zalloc().
+                   This now only enables the failed-malloc feature.
 
   enable-crypto-mdebug-backtrace
-                   As for crypto-mdebug, but additionally provide backtrace
-                   information for allocated memory.
-                   TO BE USED WITH CARE: this uses GNU C functionality, and
-                   is therefore not usable for non-GNU config targets.  If
-                   your build complains about the use of '-rdynamic' or the
-                   lack of header file execinfo.h, this option is not for you.
-                   ALSO NOTE that even though execinfo.h is available on your
-                   system (through Gnulib), the functions might just be stubs
-                   that do nothing.
+                   This is a no-op; the project uses the compiler's
+                   address/leak sanitizer instead.
```

Comment:
In `Configure`, there's the hash table `%deprecated_disablables`.  `"crypto-mdebug-backtrace"` should be removed from `@disablables` and added to `%deprecated_disablables`, like this:

``` diff
diff --git a/Configure b/Configure
index 22082deb4c..38b26b7d4f 100755
--- a/Configure
+++ b/Configure
@@ -366,7 +366,6 @@ my @disablables = (
     "cms",
     "comp",
     "crypto-mdebug",
-    "crypto-mdebug-backtrace",
     "ct",
     "deprecated",
     "des",
@@ -463,6 +462,7 @@ my @disablables_int = qw(
 my %deprecated_disablables = (
     "ssl2" => undef,
     "buf-freelists" => undef,
+    "crypto-mdebug-backtrace" => undef,
     "hw" => "hw",               # causes cascade, but no macro
     "hw-padlock" => "padlockeng",
     "ripemd" => "rmd160",
```

It could be that `"crypto-mdebug"` should be renamed to `"failed-macro"`, with deprecation of the old name.  That would look like this:

``` diff
diff --git a/Configure b/Configure
index 22082deb4c..5ee0955c5c 100755
--- a/Configure
+++ b/Configure
@@ -365,7 +365,6 @@ my @disablables = (
     "cmp",
     "cms",
     "comp",
-    "crypto-mdebug",
     "crypto-mdebug-backtrace",
     "ct",
     "deprecated",
@@ -386,6 +385,7 @@ my @disablables = (
     "engine",
     "err",
     "external-tests",
+    "failed-malloc",
     "filenames",
     "fips",
     "fuzz-libfuzzer",
@@ -463,6 +463,7 @@ my @disablables_int = qw(
 my %deprecated_disablables = (
     "ssl2" => undef,
     "buf-freelists" => undef,
+    "crypto-mdebug" => "failed-malloc",
     "hw" => "hw",               # causes cascade, but no macro
     "hw-padlock" => "padlockeng",
     "ripemd" => "rmd160",
```

You may need to check how that affects the feature macros (look in the generated include/openssl/opensslconf.h)

Reply:
I did the deprecation.  I don't want to add a new (esp new aliased!) ifdef.  Also, malloc-failing is used for debugging :)